dubbo-remoting/dubbo-remoting-api/src/main/java/com/alibaba/dubbo/remoting/transport/DecodeHandler.java

```diff
@@ -55,16 +55,11 @@ private void decode(Object message) {
             try {
                 ((Decodeable) message).decode();
                 if (log.isDebugEnabled()) {
-                    log.debug(new StringBuilder(32).append("Decode decodeable message ")
-                            .append(message.getClass().getName()).toString());
+                    log.debug("Decode decodeable message " + message.getClass().getName());
                 }
             } catch (Throwable e) {
                 if (log.isWarnEnabled()) {
-                    log.warn(
-                            new StringBuilder(32)
-                                    .append("Call Decodeable.decode failed: ")
-                                    .append(e.getMessage()).toString(),
-                            e);
+                    log.warn("Call Decodeable.decode failed: " + e.getMessage(), e);
```

Comment:
the capacity of StringBuilder in modified statement is 16, not equals with 32 in the original one

Reply:
Thanks @kimmking , what do you think about this change?

Reply:
remain original codes

Reply:
I think this is a good optimization, let's wait for other suggestions before merge.

Reply:
Thanks @chickenlj . For those cases, the naive summation will be much faster than ```StringBuilder```. I made a simple test:
```
public static void main(String[] args) {

        for (int t = 0; t < 10; t++) {
            long st = System.currentTimeMillis();

            for (int i = 0; i < 100000; i++) {
                String s = "Decode decodeable message " + "xxx";
            }

            long e = System.currentTimeMillis();
            System.out.print(e - st);
            System.out.print(" ");
            
            st = System.currentTimeMillis();

            for (int i = 0; i < 100000; i++) {
                new StringBuilder(32).append("Decode decodeable message ").append("xxx").toString();
            }
            e = System.currentTimeMillis();
            System.out.println(e - st);
        }
    }
``` and I got output:
```
1 13
0 11
0 4
0 3
0 4
0 4
0 4
0 1
0 1
0 1
```.

Reply:
this is not an equivalent comparison. the compiler may optimize `"Decode decodeable message " + "xxx"` into `"Decode decodeable message xxx"` directly.

Reply:
Thanks @beiwei30 ! That's my mistake. The test is uncorrect. The two strings cannot both be constants. 👍